packages/flutter_tools/lib/executable.dart

```diff
@@ -138,17 +137,14 @@ Future<int> run(List<String> args, List<FlutterCommand> subCommands, {
     // Initialize the system locale.
     await intl.findSystemLocale();
 
-    final Completer<int> runCompleter = new Completer<int>();
-    Chain.capture<Future<Null>>(() async {
+    try {
```

Comment:
/cc @yjbanov who had occasion to change this code to work this way just last week.